src/app/shared/components/routes/routes.component.html

```diff
@@ -1 +1,105 @@
-<p>routes works!</p>
+<div class="wrapper">
+  <div class="wrapper-forms-button">
+    <button mat-icon-button (click)="isOpen = true" [disabled]="updateMode">New route</button>
+    <button mat-icon-button (click)="isOpen = false" *ngIf="isOpen">Close</button>
+  </div>
+  <button class="reverse" mat-icon-button (click)="isReverse = !isReverse">
+    <mat-icon color="primary">autorenew</mat-icon>
+  </button>
+</div>
+
+<app-create-form (changeOpen)="isOpen = $event" *ngIf="isOpen"></app-create-form>
+
+<div class="formContainer" #formElement>
+  <ng-container *ngIf="updateMode">
+    <app-update-form
+      [currentRoute]="currentRoute"
+      (closeForm)="updateMode = $event"
+      #anchorElement
+    ></app-update-form>
+  </ng-container>
+</div>
+
+<div class="overlay" *ngIf="isLoading$ | async">
+  <mat-spinner></mat-spinner>
+</div>
+
+<div class="dialog" *ngIf="dialogOpen">
+  <h2 mat-dialog-title>Delete route</h2>
+  <div class="content">Would you like to delete route №{{ deleteId }}?</div>
+  <div class="actions">
+    <button mat-button (click)="dialogOpen = false">No</button>
+    <button mat-button (click)="onDeleteRoute(deleteId)">Yes</button>
+  </div>
+</div>
+
+<mat-divider></mat-divider>
+<div
+  class="route"
+  *ngFor="
+    let route of (isReverse ? (routes$ | async | reverse) : (routes$ | async))!.slice(
+      (NumberPage - 1) * 5,
+      NumberPage * 5
+    );
+    index as i
+  "
+>
+  <div class="route-header">
+    <h3>Route {{ route.id }}</h3>
+    <div
+      class="route-header-buttons"
+      (mousemove)="showButtonsGoal()"
+      (mouseleave)="hideButtonsGoal()"
+    >
+      <button mat-icon-button title="Update" (click)="updateRoute(route)" [disabled]="createMode">
+        <mat-icon color="primary">edit_square</mat-icon>
+        <span *ngIf="showMode">Update ride</span>
+      </button>
+      <button mat-icon-button title="Assign ride">
+        <mat-icon color="primary">train</mat-icon>
+        <span *ngIf="showMode">Assign ride</span>
+      </button>
+      <button mat-icon-button title="Delete ride" (click)="openDialog(route.id)">
+        <mat-icon color="primary">delete</mat-icon>
+        <span *ngIf="showMode">Delete ride</span>
+      </button>
+    </div>
+  </div>
+  <mat-divider></mat-divider>
+  <div class="cities">
+    <h4>Cities:</h4>
+    <span *ngFor="let city of getCities(route.path) | async; last as isLast"
+      >{{ isLast ? city : city + ' -' }}
+    </span>
+  </div>
+  <mat-divider></mat-divider>
+  <div class="carriages-types">
+    <h4>Carriages Types:</h4>
+    <span *ngFor="let type of route.carriages; last as isLast">{{
+      isLast ? type : type + ' -'
+    }}</span>
+  </div>
+</div>
+<div class="pagination">
```

Comment:
Perfect implementation of custom pagination!
Just for information, angular material has paginator https://material.angular.io/components/paginator/overview